doc/coq.rst

```diff
@@ -142,19 +142,17 @@ The semantics of the fields are:
   You may still use installed libraries in your Coq project, but there is
   currently no way for Dune to know about it.
 
-- You can enable the production of Coq's native compiler object files by setting
-  ``<coq_native_mode>`` to ``native``. This passes ``-native-compiler on`` to
-  Coq and install the corresponding object files under ``.coq-native``, when in
-  the ``release`` profile. The regular ``dev`` profile skips native compilation
-  to make the build faster. This has been available since :ref:`Coq lang
-  0.3<coq-lang>`.
-
-  Please note: support for ``native_compute`` is **experimental** and requires a
-  version of Coq later than 8.12.1. Furthermore, dependent theories *must* be
-  built with the ``(mode native)`` enabled. In addition to that, Coq must be
-  configured to support native compilation. Dune explicitly disables the
-  generation of native compilation objects when ``(mode vo)`` is enabled,
-  irrespective of the configuration of Coq. This will be improved in the future.
+- From version :ref:`Coq lang 0.7<coq-lang>` onwards, if Coq has been configured
+  with ``-native-compiler yes`` or ``ondemand``, Dune will always build the
+  ``cmxs`` files together with the ``vo`` files.
+  
+  You may override this by specifying ``(mode native)`` or ``(mode vo)``. Before
+  :ref:`Coq lang 0.7<coq-lang>`, the native mode had to be manually specified.
+
+  When developing with native, you may want the compilation to be faster. You
+  may add ``(mode vo)`` to your stanza to override the native mode and only
+  compile ``vo`` files. Previous versions of Dune before 3.6 would disable the
+  native rules depending on whether or not the ``dev`` profile was selected.
```

Comment:
Umm, this seems a bit bizarre, certainly is not standard dune practice to have to edit an stanza when developing.

I would just state that `(mode vo)` disables native compilation even if the configure option is set.

Reply:
We already have that `(mode)` overrides the configure option. Perhaps I should just mention using a native opam switch or something. Though, I am not certain how this works so I will just remove it all together and leave it to a future doc improvement.